File: src/main/java/com/unicolour/joyspace/service/ManagerService.java
package com.unicolour.joyspace.service;

import com.unicolour.joyspace.model.Manager;
import org.springframework.security.core.userdetails.UserDetailsService;

public interface ManagerService extends UserDetailsService {
	Manager createManager(String userName, String password, String fullName, String cellPhone);
	Manager editUser(int id, String fullName, String cellPhone);
	boolean resetPassword(int userId, String password);
	Manager login(String userName, String password);
}

File: src/main/kotlin/com/unicolour/joyspace/service/ManagerService.kt
package com.unicolour.joyspace.service

import com.unicolour.joyspace.dto.LoginManagerDetail
import com.unicolour.joyspace.dto.WxLoginResult
import com.unicolour.joyspace.model.Company
import com.unicolour.joyspace.model.Manager
import org.springframework.security.core.userdetails.UserDetailsService

interface ManagerService : UserDetailsService {
    fun createManager(userName: String, password: String, fullName: String, cellPhone: String,
                      email: String, roles: String, company: Company): Manager
    fun resetPassword(userId: Int, password: String): Boolean
    fun login(userName: String, password: String): Manager?
    val loginManager: LoginManagerDetail?
    fun loginManagerHasRole(role: String): Boolean

    fun getCompanyManager(companyId: Int): Manager?
}
